lnwire/closing_complete.go

```diff
@@ -0,0 +1,156 @@
+package lnwire
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+// ClosingSigs houses the 3 possible signatures that can be sent when
+// attempting to complete a cooperative channel closure. A signature will
+// either include both outputs, or only one of the outputs from either side.
+type ClosingSigs struct {
+	// CloseNoClosee is a signature that excludes thte output of the
+	// clsoee.
+	CloserNoClosee tlv.OptionalRecordT[tlv.TlvType1, Sig]
+
+	// NoCloserClosee is a signature that excludes the output of the
+	// closer.
+	NoCloserClosee tlv.OptionalRecordT[tlv.TlvType2, Sig]
+
+	// CloserAndClosee is a signature that includes both outputs.
+	CloserAndClosee tlv.OptionalRecordT[tlv.TlvType3, Sig]
+}
+
+// ClosingComplete is sent by either side to kick off the process of obtaining
+// a valid signature on a c o-operative channel closure of their choice.
+type ClosingComplete struct {
+	// ChannelID serves to identify which channel is to be closed.
+	ChannelID ChannelID
+
+	// FeeSatoshis is the total fee in satoshis that the party to the
+	// channel would like to propose for the close transaction.
+	FeeSatoshis btcutil.Amount
+
+	// Sequence is the sequence number to be used in the input spending the
+	// funding transaction.
+	Sequence uint32
+
+	// ClosingSigs houses the 3 possible signatures that can be sent.
+	ClosingSigs
+
+	// ExtraData is the set of data that was appended to this message to
+	// fill out the full maximum transport message size. These fields can
+	// be used to specify optional data such as custom TLV fields.
+	ExtraData ExtraOpaqueData
+}
+
+// decodeClosingSigs decodes the closing sig TLV records in the passed
+// ExtraOpaqueData.
+func decodeClosingSigs(c *ClosingSigs, tlvRecords ExtraOpaqueData) error {
+	sig1 := tlv.ZeroRecordT[tlv.TlvType1, Sig]()
```

Comment:
Can prob make a `Zero()` method on the record type to abstract this a bit. Then this func doesn't need to know the actual record type (`tlv.TlvType2`) by name.

Reply:
What would a `Zero()` method look like?

Reply:
Something like this: 
```go
func (t *Record[T, V]) Zero() Record[T, V] {
       var v V
       return Record[T, V] {
               Val: v, 
      } 
}
```

Then this code area would look like: 
```go
sig1 := c.CloserNoClosee.Zero()
```

etc